core/__tests__/core.test.ts

```diff
@@ -0,0 +1,84 @@
+import Factory from '../src/';
+import {simpleLogger, simpleErrorHandler} from '../src/defaults';
+
+describe('AsyncStorageFactory', () => {
+  it('Throws when tried to instantiate', () => {
+    expect(() => new Factory()).toThrow();
```

Comment:
please add some text that you expect to throw, it may have thrown for other reasons and test will pass, something like:
```suggestion
    expect(() => new Factory()).toThrow('forbidden to instantiate Factory');
```